diff --git a/Source/Core/Common/StringUtil.cpp b/Source/Core/Common/StringUtil.cpp
index 935db75d2f..ad5bf07405 100644
--- a/Source/Core/Common/StringUtil.cpp
+++ b/Source/Core/Common/StringUtil.cpp
@@ -487,6 +487,20 @@ std::string UTF16BEToUTF8(const char16_t* str, size_t max_size)
 template <typename T>
 std::string CodeTo(const char* tocode, const char* fromcode, std::basic_string_view<T> input)
 {
+
+  char* locale_res = setlocale(LC_ALL, "");
+  if (!locale_res)
+  {
+    ERROR_LOG_FMT(COMMON,
+                  "setlocale(LC_ALL, \"\") failed [{}]: {}, falling back to C locale",
+                  fromcode, strerror(errno));
+    if (!setlocale(LC_ALL, "C"))
+    {
+        ERROR_LOG_FMT(COMMON,
+                      "setlocale(LC_ALL, \"C\") failed: {}",
+                      strerror(errno));
+    }
+  }
   std::string result;
 
   iconv_t const conv_desc = iconv_open(tocode, fromcode);
